### ISSUE TITLE
test: ext_watchman.php regression

### ISSUE THREAD

**Issue by cmuellner** [facebook/hhvm#7649]
current git master fails to run ext_watchman.php test
on (at least) aarch64 Ubuntu Xenial (see [1]). It is assumed that
recent changes of the watchman code introduced this
regression.

The build has been done using this commands:

```
rm -f ~/.hhvm.hhbc
git clean -dfx
rm -rf third-party
git submodule update --init --recursive
cmake -DENABLE_LD_GOLD=Off .
make -j8
./hphp/hhvm/hhvm hphp/test/run all
```

The test diff show the following:

```
FAILED: hphp/test/slow/ext_watchman/ext_watchman.php
001+ Possible bad config node: watchman
002+ 
003+ Fatal error: Call to undefined function HH\ext_watchman_version() in /mnt/cmuellner/hhvm/hphp/test/slow/ext_watchman/ext_watchman.php on line 137
001- PASSED version get test
002- Testing one\-shot
003- PASS
004- Testing subscription
005- Subscribed
006- Waiting for got_fresh
007- PASS
008- Waiting for got_update
009- PASS
010- Unsubscribing
011- PASS
012- Confirming subscription no longer exists
013- PASS
014- Test corrupt subscription callback
015- Got expected exception\: Problem\(s\) on subscription named 'X'\. Connection IS alive\.
016- First error\:
017- syntax error, unexpected '\}'
018- PASS
019- Test delete subscription callback
020- Got expected exception\: Problem\(s\) on subscription named 'X'\. Connection IS alive\.
021- First error\:
022- [^\r\n]+\: Unit '\/tmp\/[^\r\n]+' no longer exists\.
023- PASS
024- Stess test subscription
025- \.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.
026- PASS \(\d+, \d+, \d+, \d+\)
027- Testing syncing
028- Subscribed, sleeping for 1 second\.\.\.
029- Sync expecting timeout after 100ms\.\.\.
030- PASS
031- Sync expecting no timeout\.\.\.
032- PASS
033- Sync expecting immediate return\.\.\.
034- PASS
035- Unsubscribing
036- Check catching exception raised in a subscription callback
037- Got exception\: Problem\(s\) on subscription named 'X'\. Connection IS alive\.
038- First error\:
039- exception 'Exception' with message 'expected exception' in [^\r\n]+\:\d+
040- Stack trace\:
041- #0 \(\)\: callback_exception\(\)
042- #1 \{main\}
043- PASS
044- Check unsubscribing while callback in progress
045- PASS
046- Checking connecting to a bad socket is graceful
047- Got exception\: [^\r\n]+AsyncSocketException[^\r\n]+
048- PASS
049- Checking detection of dead Watchman server connection
050- PASS \(connection error\)
```

[1] http://hhvm-arm64.theobroma-systems.com/1486429201482206496/test.txt

**Comment by cmuellner** [comment]
The following systems are confirmed to be affected by this bug:

* x86_64 CentOS 7
* x86_64 Fedora 25
* aarch64 Ubuntu Xenial
* aarch64 Fedora 25
* aarch64 CentOS 7
* aarch64 OpenSuse LEAP 42.2

Further debug and release builds are affected.

**Comment by swalk-cavium** [comment]
I see this test fail for interp, jit -pgo, and jit +pgo option sets on Aarch64 Ubuntu.

**Comment by cmuellner** [comment]
Mentioning @paulbiss and @markw65 as I don't know the github username of Jacob Bower <jbower@fb.com> (who seems to have a good expertise in the ext_watchman code).

**Comment by cjiph** [comment]
@cmuellner, thanks for catching this. I added ext_watchman and I realize now we should exclude this test for systems which don't have Watchman installed/configured at build time. I will see if I can get this resolved shortly.

**Comment by cmuellner** [comment]
@swalk-cavium fixed this via #7840 (pulled by mofarrell in acbdaa3). Thanks for that!
